pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/SourceIdentifierScannerTest.scala

```diff
@@ -0,0 +1,190 @@
+package uk.ac.wellcome.platform.idminter.steps
+
+import org.scalatest.{FunSpec, Matchers}
+import uk.ac.wellcome.models.work.generators.IdentifiersGenerators
+import io.circe.generic.auto._
+import io.circe.syntax._
+import io.circe.parser._
+import uk.ac.wellcome.models.work.internal.SourceIdentifier
+import uk.ac.wellcome.platform.idminter.fixtures.SqlIdentifiersGenerators
+import uk.ac.wellcome.platform.idminter.models.Identifier
+
+import scala.util.{Failure, Success}
+
+class SourceIdentifierScannerTest
+    extends FunSpec
+    with IdentifiersGenerators
+    with Matchers
+    with SqlIdentifiersGenerators {
+  case class ClassWithIdentifier(
+    sourceIdentifier: SourceIdentifier,
+    classWithIdentifiers: List[ClassWithIdentifier] = Nil)
+  case class ClassWithIdentifierAndIdentifiedType(
+    sourceIdentifier: SourceIdentifier,
+    identifiedType: String,
+    classWithIdentifiers: List[ClassWithIdentifierAndIdentifiedType] = Nil
+  )
+  case class ClassWithIdentifierAndCanonicalId(
+    sourceIdentifier: SourceIdentifier,
+    canonicalId: String,
+    classWithIdentifiers: List[ClassWithIdentifierAndCanonicalId] = Nil)
+  case class ClassWithIdentifierTypeAndCanonicalId(
+    sourceIdentifier: SourceIdentifier,
+    canonicalId: String,
+    `type`: String,
+    classWithIdentifiers: List[ClassWithIdentifierTypeAndCanonicalId] = Nil
+  )
+  describe("scan") {
+    it("retrieves a sourceIdentifiers at the root of the json") {
+      val sourceIdentifier = createSourceIdentifier
+      val objectWithIdentifier = ClassWithIdentifier(sourceIdentifier)
+      SourceIdentifierScanner
+        .scan(objectWithIdentifier.asJson)
+        .get shouldBe List(sourceIdentifier)
```

Comment:
It's more lines, but IMO these tests would be way more readable if I could see the JSON strings that were being manipulated directly, rather than having to memorise the case class hierarchy. For example:

```suggestion
      val jsonString = """
        |{
        |  "sourceIdentifier": {
        |    "identifierType": "miro-image-number",
        |    "value": "A0001234",
        |    "ontologyType": "Work"
        |  }
        |}
      """.stripMargin

      val sourceIdentifier = SourceIdentifier(
        identifierType = "miro-image-number",
        value = "A0001234",
        ontologyType = "Work"
      )
      
      val json = parse(jsonString).get

      SourceIdentifierScanner
        .scan(json)
        .get shouldBe List(sourceIdentifier)
```